superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts

```diff
@@ -121,10 +129,12 @@ export default function transformProps(
   }: EchartsMixedTimeseriesFormData = { ...DEFAULT_FORM_DATA, ...formData };
 
   const colorScale = CategoricalColorNamespace.getScale(colorScheme as string);
-  const rawSeriesA = extractTimeseriesSeries(rebaseTimeseriesDatum(data1), {
+  const rebasedDataA = rebaseTimeseriesDatum(data1, extractedDataSource);
```

Comment:
Does this really work? Looking at the code, I think we should be passing `verboseMap`, not `datasource`: https://github.com/apache/superset/blob/04e3cfa6077faa84b07948fc6071e94d82a4a000/superset-frontend/plugins/plugin-chart-echarts/src/utils/prophet.ts#L112-L115

So seems this is the correct solution:
```suggestion
  const rebasedDataA = rebaseTimeseriesDatum(data1, extractedDataSource?.verboseMap);
```

Reply:
@villebro This leads to this typescript error `Property 'verboseMap' does not exist on type string` due to the following definition of `datasource`. Leading back to where Its type must be changed to an `Object`.
![data](https://user-images.githubusercontent.com/42173629/145667892-904a7c4c-5dec-422a-83e0-9a635c008479.PNG)

Reply:
@Yahyakiani oh, ok, now I see. Let me take a new look, it appears some type/interface declarations are incorrect somewhere after all.

Reply:
@Yahyakiani Ok, it seems there was an error in the original `transformProps` signature; `chartProps` isn't `EchartsMixedTimeseriesFormData`, but `EchartsMixedTimeseriesProps`. After changing that and making some modifications this should probably work (I didn't test, but it seems to pass lint):

```diff
diff --git a/superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts b/superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts
index 0f0fc2a72a..c287ab90c7 100644
--- a/superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts
+++ b/superset-frontend/plugins/plugin-chart-echarts/src/MixedTimeseries/transformProps.ts
@@ -33,6 +33,7 @@ import {
   DEFAULT_FORM_DATA,
   EchartsMixedTimeseriesFormData,
   EchartsMixedTimeseriesChartTransformedProps,
+  EchartsMixedTimeseriesProps,
 } from './types';
 import { ForecastSeriesEnum } from '../types';
 import { parseYAxisBound } from '../utils/controls';
@@ -63,7 +64,7 @@ import {
 import { TIMESERIES_CONSTANTS } from '../constants';

 export default function transformProps(
-  chartProps: EchartsMixedTimeseriesFormData,
+  chartProps: EchartsMixedTimeseriesProps,
 ): EchartsMixedTimeseriesChartTransformedProps {
   const {
     width,
@@ -74,11 +75,11 @@ export default function transformProps(
     filterState,
     datasource,
   } = chartProps;
+  const { verboseMap = {} } = datasource;
   const { annotation_data: annotationData_ } = queriesData[0];
   const annotationData = annotationData_ || {};
-  const extractedDataSource = datasource || {};
-  const data1: TimeseriesDataRecord[] = queriesData[0].data || [];
-  const data2: TimeseriesDataRecord[] = queriesData[1].data || [];
+  const data1 = (queriesData[0].data || []) as TimeseriesDataRecord[];
+  const data2 = (queriesData[1].data || []) as TimeseriesDataRecord[];

   const {
     area,
@@ -129,14 +130,18 @@ export default function transformProps(
   }: EchartsMixedTimeseriesFormData = { ...DEFAULT_FORM_DATA, ...formData };

   const colorScale = CategoricalColorNamespace.getScale(colorScheme as string);
-  const rebasedDataA = rebaseTimeseriesDatum(data1, extractedDataSource);
-  const rawSeriesA = extractTimeseriesSeries(rebasedDataA, {
-    fillNeighborValue: stack ? 0 : undefined,
-  });
-  const rebasedDataB = rebaseTimeseriesDatum(data2, extractedDataSource);
-  const rawSeriesB = extractTimeseriesSeries(rebasedDataB, {
-    fillNeighborValue: stackB ? 0 : undefined,
-  });
+  const rawSeriesA = extractTimeseriesSeries(
+    rebaseTimeseriesDatum(data1, verboseMap),
+    {
+      fillNeighborValue: stack ? 0 : undefined,
+    },
+  );
+  const rawSeriesB = extractTimeseriesSeries(
+    rebaseTimeseriesDatum(data2, verboseMap),
+    {
+      fillNeighborValue: stackB ? 0 : undefined,
+    },
+  );

   const series: SeriesOption[] = [];
   const formatter = getNumberFormatter(contributionMode ? ',.0%' : yAxisFormat);
```